File: dotnet/GHIElectronics.Due_/Touch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GHIElectronics.DUE {
    public partial class DUEController {


        public class TouchController {

            SerialInterface serialPort;

            public TouchController(SerialInterface serialPort) => this.serialPort = serialPort;

            public bool Read(int pin) {
                var cmd = string.Format("print(touchread({0}))", pin);
                this.serialPort.WriteCommand(cmd);

                var res = this.serialPort.ReadRespone();
                var val = false;
                if (res.success) {


                    try {
                        val = int.Parse(res.respone) == 1 ? true : false;
                    }
                    catch { }
                }

                return val;

            }

           
            
        }
    }
}

File: python/DUELink/Touch.py
class TouchController:
    def __init__(self, serialPort):
        self.serialPort = serialPort

    def Read(self, pin: int, charge_t: int, charge_s: int, timeout: int):
        cmd = "touch({0}, {1}, {2}, {3})".format(pin, charge_t, charge_s, timeout)
        self.serialPort.WriteCommand(cmd)

        res = self.serialPort.ReadResponse()
        
        val = False
        if res.success:
            try:
                return res.response
            except:
                pass
        return val
